packages/plugin-core/src/features/RecentWorkspacesTreeview.ts

```diff
@@ -0,0 +1,75 @@
+import { DendronTreeViewKey, VSCodeEvents } from "@dendronhq/common-all";
+import { MetadataService } from "@dendronhq/engine-server";
+import * as vscode from "vscode";
+import {
+  ProviderResult,
+  TreeDataProvider,
+  TreeItem,
+  TreeItemCollapsibleState,
+  Uri,
+} from "vscode";
+import {
+  InstrumentedWrapperCommand,
+  InstrumentedWrapperCommandArgs,
+} from "../commands/InstrumentedWrapperCommand";
+
+type DendronWorkspaceMenuItem = {
+  fsPath: string;
+};
+
+/**
+ * Data provider for the Recent Workspaces Tree View
+ */
+class RecentWorkspacesTreeDataProvider
+  implements TreeDataProvider<DendronWorkspaceMenuItem>
+{
+  getTreeItem(element: DendronWorkspaceMenuItem): TreeItem {
+    const commandArgs: InstrumentedWrapperCommandArgs = {
+      command: {
+        title: "Open Workspace",
+        command: "vscode.openFolder",
+        arguments: [Uri.file(element.fsPath)],
+      },
+      event: VSCodeEvents.RecentWorkspacesPanelUsed,
+    };
+
+    return {
+      label: element.fsPath,
```

Comment:
in terms of ergonomics, the whole path to the note overflows the sidebar which is not that great: https://www.loom.com/i/b8d556dc80584b0da3caf2b2b8073c1f
can we list vaults by their names instead (and tooltip to the full path)?

Reply:
I considered this but thought it'd be more informative to a new user to list the file path as opposed to the vault name - the main intention here is to make sure new users can get back into Dendron if they had closed it out. Having this be a convenience to opening vaults/workspaces for experience users is a secondary goal.